docs/content/eventing/_index.md

```diff
@@ -1,17 +1,143 @@
+
+In Ensign, to establish a client, you need an [API key](#api-key).
+
+```golang
+import 	(
+    ensign "github.com/rotationalio/ensign/sdks/go"
+)
+
+client, _ := ensign.New(&ensign.Options{
+	ClientID: "FMDmvoZeMrLxUTJVaDbIxBEtIUgNICln",
+	ClientSecret: "oeVJghlSIK2YbMqOMCUiSVRVQOLT0ORrVSwAfRpXLTiWn7yo7HQzOCwxMvveqiHX",
+})
+```
+
+#### **event** <a name="events"></a>
+In an event-driven or [microservice](#microservice) architecture, an event is the atomic element of data.
+
+An event might look something like a dictionary, which is then wrapped in an object or struct that provides some schema information to help Ensign know how to serialize and deserialize your data.
+
+```golang
+order := make(map[string]string)
+order["item"] = "large mushroom pizza"
+order["customer_id"] = "984445"
+order["customer_name"] = "Enson J. Otterton"
+order["timestamp"] = time.Now().String()
+
+evt := &api.Event{
+    TopicId:  "order-feed",
+    Mimetype: mimetype.ApplicationJSON,
+    Type: &api.Type{
+        Name:    "Generic",
+        Version: 1,
+    },
+}
+
+evt.Data, _ = json.Marshal(order)
+```
+
+#### **latency** <a name="latency"></a>
+Latency can refer to both application-level communication lag (e.g. the time it takes for one part of the code to finish running before moving on to the next part) or to network communication lag (e.g. the time it takes for two remote servers on two different continents to send a single message back and forth).
+
+Less latency is better, generally speaking.
+
+In a microservices context, we can reduce application latency by using asynchronous communications and parallelizing functions so they run faster. Network latency can be reduced by creating more efficient communications between servers (e.g. using more scalable consensus algorithms).
+
+
+#### **microservice** <a name="microservice"></a>
+A microservice is a computer application composed of a collection of lightweight services, each of which is responsible for some discrete task.
+
+Microservices can be coordinated to communicate via [events](#events).
+
+
+#### **mime type**
+A MIME (Multipurpose Internet Mail Extensions) type is a label that identifies a type of data, such as CSV, HTML, JSON, or protocol buffer.
+
+MIME types allow an application to understand how to handle incoming and outgoing data.
+
+
+#### **organization**
+An Ensign organization is a collection of users who are working under the same Ensign [tenant](#tenant).
+
+
+#### **publisher** <a name="publisher"></a>
+In an event-driven microservice, a publisher is responsible for emitting [events](#events) to a [topic stream](#topic).
+
+In Ensign, you can create a publisher once you have established a [client](#client):
+
+```golang
+pub, _ := client.Publish(...)
```

Comment:
Similarly should we specify what the other field being returned is or leave it to the API docs?